projects/igniteui-angular/src/lib/combo/combo.component.spec.ts

```diff
@@ -2286,6 +2286,43 @@ describe('igxCombo', () => {
                 }, 20);
             }, 10);
         });
+
+        it('Should bind combo data to remote data and clear selection properly when items are out of view', async (done) => {
+            const fixture = TestBed.createComponent(IgxComboRemoteDataComponent);
+            fixture.detectChanges();
+            const combo = fixture.componentInstance.instance;
+            const selectedItems = [combo.data[0], combo.data[1]];
+            combo.toggle();
+            combo.selectItems([selectedItems[0][combo.valueKey], selectedItems[1][combo.valueKey]]);
+            expect(combo.value).toEqual(`${selectedItems[0][combo.displayKey]}, ${selectedItems[1][combo.displayKey]}`);
+            expect(combo.selectedItems()).toEqual([selectedItems[0][combo.valueKey], selectedItems[1][combo.valueKey]]);
+            combo.virtualScrollContainer.scrollTo(40);
+            await wait(60);
+            fixture.detectChanges();
+            const spyObj = jasmine.createSpyObj('event', ['stopPropagation']);
+            combo.handleClearItems(spyObj);
+            expect(combo.selectedItems().indexOf(selectedItems[0][combo.valueKey]) < 0).toBeTruthy();
+            expect(combo.selectedItems().indexOf(selectedItems[1][combo.valueKey]) < 0).toBeTruthy();
+            expect(combo.value).toBe('');
```

Comment:
You are missing a select on one of the items in view, to verify the state of the selection is correct then.